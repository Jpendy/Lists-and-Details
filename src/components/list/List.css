.nameTag {
    font-size: 20px;
}

.image {
    width: 300px;
    border-radius: 20px 80px 20px 80px;
    box-shadow: 1px 2px 10px 2px;
}

.listItem {
    margin: 50px auto;
    text-align: center;
    list-style: none;
}