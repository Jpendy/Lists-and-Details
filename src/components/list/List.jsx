import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const List = ({ listArray }) => {

  const listElement = listArray.map(item => {
    return (
      <Link key={item._id} to={`/${item.name}`} >
          
        <li>
          <p>{item.name}</p>
          <img src={item.photoUrl} />
        </li>
      </Link>
    );
  });
  
  return (
    <ul>
      { listElement }
    </ul>
  );
};

List.propTypes = {
  listArray: PropTypes.arrayOf(PropTypes.object.isRequired).isRequired
};

export default List;
