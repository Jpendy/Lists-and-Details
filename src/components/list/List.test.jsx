import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

describe('List component', () => {
  it('renders List', () => {
    const wrapper = shallow(<List listArray={[{ thing: 'thing1' }, { thing: 'thing2' }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
