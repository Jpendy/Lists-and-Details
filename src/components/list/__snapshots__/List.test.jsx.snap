// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List 1`] = `
<ul>
  <Link
    key="1"
    to="/undefined"
  >
    <li>
      <p />
      <img />
    </li>
  </Link>
  <Link
    key="2"
    to="/undefined"
  >
    <li>
      <p />
      <img />
    </li>
  </Link>
</ul>
`;
